lib/ledger_sync/adaptors/quickbooks_online/ledger_serializer_type/transaction_reference_type.rb

```diff
@@ -0,0 +1,58 @@
+# frozen_string_literal: true
+
+module LedgerSync
+  module Adaptors
+    module QuickBooksOnline
+      module LedgerSerializerType
+        class TransactionReferenceType < Adaptors::LedgerSerializerType::ValueType
+          def convert_from_ledger(value:)
+            return if value.nil?
+
+            raise "Unknown value type.  Array expected.  Given: #{value.class.name}" unless value.is_a?(Array)
+            return if value.empty?
+
+            value.map do |item|
+              resource_class = begin
+                quickbooks_online_type_hash = QuickBooksOnline::LedgerSerializer.quickbooks_online_resource_types_hash.fetch(item['TxnType'].downcase, nil)
+                if quickbooks_online_type_hash.present?
+                  quickbooks_online_type_hash.try(:fetch, :resource_class, nil)
+                else
+                  LedgerSync.resources[item['TxnType'].downcase.to_sym]
+                end
+              end
+
+              raise "Unknown QuickBooks Online resource type: #{item['TxnType']}" if resource_class.blank?
+
+              ret = resource_class.new(
+                ledger_id: item['TxnId']
+              )
+            end
+          end
+
+          def convert_from_local(value:)
+            return if value.nil?
+            raise "List expected.  Given: #{value.class.name}" unless value.is_a?(Array)
+            # raise "Resources expected.  Given: #{value.map{|i| i.class.name}.join(', ')}" unless value.map{|i| i.class}.all?(LedgerSync::Resource)
```

Comment:
So this was the interesting part. I basically took this from another type you wrote. And you compare that object is a `LedgerSync::Resource`, but in this case it was failing, because object is `LedgerSync::Invoice`. So wanted to run this by you.